src/webview/js/handleInboundEvents.js

```diff
@@ -0,0 +1,44 @@
+/* @flow strict-local */
+import invariant from 'invariant';
+import { ensureUnreachable } from '../../generics';
+
+import type { EditSequence } from '../generateInboundEventEditSequence';
+
+// TODO: Put more of the WebView's event-handling code in here.
+
+/**
+ * PRIVATE - don't use until we have more test coverage (coming soon)
+ */
+export function applyEditSequence(editSequence: EditSequence) {
+  const msglistElementsDiv = document.querySelector('div#msglist-elements');
+  invariant(msglistElementsDiv, 'applyEditSequence: expected msglistElementsDiv');
+
+  editSequence.forEach(edit => {
+    switch (edit.type) {
+      case 'insert': {
+        const fragment = document.createDocumentFragment();
+        const newChild = document.createElement('div');
+        fragment.appendChild(newChild);
+        newChild.outerHTML = edit.html;
+        msglistElementsDiv.insertBefore(fragment, msglistElementsDiv.children[edit.index]);
```

Comment:
Hmm, what does this do if `edit.index` is just past the end of the list? E.g., if it's zero and the list is empty.

That second argument will be `undefined`, presumably. And MDN has a warning that that's no good -- you need to pass `null` for that argument instead:
https://developer.mozilla.org/en-US/docs/Web/API/Node/insertBefore

The links there indicate that at least Chrome throws an exception if it's missing/undefined, and is following the spec by doing so. Seems like jsdom has a bug that it doesn't do the same.

Reply:
Oho. No, this is subtle, and MDN is misleading there. In fact, you can pass `undefined` there and it behaves the same as passing `null`. What you can't do is pass *no* argument.

Typically in JS passing a final argument `undefined` is just like not passing that argument at all. But in JS bindings of a WebIDL interface, such as the DOM, a parameter which the IDL says is a nullable type actually accepts `undefined`, and converts it to the IDL `null`, just like it would the JS `null`:
https://webidl.spec.whatwg.org/#es-nullable-type

On the other hand if no argument is passed there at all, then that method ("operation") simply isn't found in the first place, in the "overload resolution algorithm":
https://webidl.spec.whatwg.org/#dfn-create-operation-function
Effectively it's like a pure-JS function that inspects its `arguments.length` and errors out if it's too short.

Anyway, so this code is fine per the spec, and jsdom is behaving correctly with it.

Reply:
Aha, weird! I think I had yet another interpretation of what was going on. 😛

I think I assumed `msglistElementsDiv.children[edit.index]` would be `null` when `edit.index` is out of bounds—like [`msglistElementsDiv.children.item(edit.index)`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCollection#methods) would be.

I knew `.children` didn't give a straightforward JavaScript `Array`, but in retrospect I probably shouldn't have taken that to mean the usual `myArr[i]` syntax would behave specially (giving `null` for out-of-bounds instead of the usual `undefined`).

Reply:
I guess I'm assuming MDN is correct that `msglistElementsDiv.children[edit.index]` _doesn't_ give `null` for out-of-bounds; [link](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCollection#methods):

> An alternative to accessing `collection[i]` (which instead returns `undefined` when `i` is out-of-bounds). This is mostly useful for non-JavaScript DOM implementations.

Reply:
> this is subtle

Definitely. Maybe the path of least needed explanation is to use `msglistElementsDiv.children.item(edit.index)`? Then we don't have to give a code comment about why MDN is wrong about how `Node.insertBefore()` behaves when you pass `undefined` for the second argument.

Reply:
That or `?? null`, which might be still more transparent for readers familiar with JS but not as much with the details of the DOM.